src/mlpack/methods/ann/layer/max_pooling.hpp

```diff
@@ -168,21 +170,39 @@ class MaxPoolingType : public Layer<MatType>
       for (size_t j = 0, colidx = 0; j < output.n_cols;
           ++j, colidx += strideHeight)
       {
+        size_t colEnd = colidx + kernelHeight - 1;
+        // Check if the kernel along column is out of bounds.
+        if (colEnd > input.n_cols - 1)
+        {
+          // If so, we need to reduce the kernel height or terminate.
+          if (floor)
+            continue;
```

Comment:
nit: I think the style guides suggest using braces since they look cleaner and are more readable.

Reply:
According to Design Docs:
To conserve lines, refrain from using braces if the `if`, `while`, or `for` statement applies to a single statement.